### PR TITLE
[DPR2-494] Add DPD schema and validate DPDs

### PR DESCRIPTION
**Pull request by stalwartsolutions**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/dprdpdsservice/controller/integration/SchemaTest.kt
Comment (gavriil-g-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/dprdpdsservice/controller/integration/SchemaTest.kt:24]: Does this path need to get printed?

File: src/main/resources/data-product-definition-schema.json
Comment (gavriil-g-moj) [inline src/main/resources/data-product-definition-schema.json:237]: Is this schema meant to be the same as https://github.com/ministryofjustice/hmpps-digital-prison-reporting-lib/blob/main/src/main/resources/data-product-definition-schema.json?
There are some divergences between them. For example the "type" in the library schema also has "null', "bytes" and "timestamp". 
Also "wordwrap" in the library has "on", "off", "none".
And "visible" in the library is outdated. 
I suggest after we bring the format up to date to keep only one schema file in one repository to avoid one of them falling behind.
It might be better to keep it in the library as clients would also be interested in this schema file. 
And in the definitions repository to put a link in the Readme. 
But since it is used in the test for the definitions validation this might be a problem to remove it.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/dprdpdsservice/controller/integration/SchemaTest.kt
Comment (stalwartsolutions) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/dprdpdsservice/controller/integration/SchemaTest.kt:24]: I think it's helpful to see that the files have been checked.

File: src/main/resources/data-product-definition-schema.json
Comment (stalwartsolutions) [reply src/main/resources/data-product-definition-schema.json:237]: In terms of those changes to the schema, yes that's exactly right:
Screenshot 2024-03-22 at 08 35 04

I'm just waiting on Dominic to confirm before I merge these changes.

Comment (stalwartsolutions) [reply src/main/resources/data-product-definition-schema.json:237]: In terms of the library being outdated; I think the other PR should solve all of these.

**Review by gavriil-g-moj — APPROVED** [review]
I have approved the PR so that these changes are not held up and we can decide later where the best place if for the json schema to live.

**Pull request merged**
